src/main/java/net/codejava/MvcController.java

```diff
@@ -69,12 +70,21 @@ private void updateAccountInfo(User user) {
     String getUserNameAndBalanceAndOverDraftBalanceSql = String.format("SELECT FirstName, LastName, Balance, OverdraftBalance FROM customers WHERE CustomerID='%s';", user.getUsername());
     List<Map<String,Object>> queryResults = jdbcTemplate.queryForList(getUserNameAndBalanceAndOverDraftBalanceSql);
     String getOverDraftLogsSql = String.format("SELECT * FROM OverdraftLogs WHERE CustomerID='%s';", user.getUsername());
+    String getTransactionHistorySql = String.format("Select * from TransactionHistory WHERE CustomerId='%s';", user.getUsername());
     
     List<Map<String,Object>> queryLogs = jdbcTemplate.queryForList(getOverDraftLogsSql);
     String logs = HTML_LINE_BREAK;
-    for(Map<String, Object> x : queryLogs){
-      logs += x + HTML_LINE_BREAK;
+    for(Map<String, Object> overdraftLog : queryLogs){
+      logs += overdraftLog + HTML_LINE_BREAK;
     }
+    List<Map<String,Object>> transLogs = jdbcTemplate.queryForList(getTransactionHistorySql);
+    String transactionHistory = HTML_LINE_BREAK;
+    int numLog = 0;
+    for(Map<String, Object> transHist : transLogs){
+      if(numLog >= transLogs.size() - 3) transactionHistory += transHist + HTML_LINE_BREAK;

```

Comment:
A better way to do this would be to limit the SQL query to only return the 3 most recent transaction logs for this customer. With this approach, you are iterating through every transaction log of the customer, and also assuming that the output comes out in the appropriate order such that the final three rows returned are the most recent.

I understand that SQL is not a pre-req for this class, or taught in this class, so I will take care of pushing a fix for this.

Also, never do an in-line if-statement + body. It's not that readable, and a fringe feature of the Java language.